external-crates/move/move-compiler/src/typing/core.rs

```diff
@@ -368,40 +357,18 @@ impl<'env> Context<'env> {
         constants.get(n).expect("ICE should have failed in naming")
     }
 
-    pub fn in_loop(&self) -> bool {
-        match &self.loop_info.0 {
-            LoopInfo_::NotInLoop => false,
-            LoopInfo_::BreakTypeUnknown | LoopInfo_::BreakType(_) => true,
-        }
-    }
-
-    pub fn get_break_type(&self) -> Option<&Type> {
-        match &self.loop_info.0 {
-            LoopInfo_::NotInLoop | LoopInfo_::BreakTypeUnknown => None,
-            LoopInfo_::BreakType(t) => Some(t),
-        }
-    }
-
-    pub fn set_break_type(&mut self, t: Type) {
-        match &self.loop_info.0 {
-            LoopInfo_::NotInLoop => (),
-            LoopInfo_::BreakTypeUnknown | LoopInfo_::BreakType(_) => {
-                self.loop_info.0 = LoopInfo_::BreakType(Box::new(t))
-            }
+    pub fn named_block_type(&mut self, name: Var) -> Type {
```

Comment:
Where does the `loc` come from for the `Var`? 
I think it is from the declaration of the loop right? 
We should pass in the `loc` so that we can set it to the first `break` when applicable